app/models/commit.rb

```diff
@@ -88,7 +88,12 @@ def add_status(status_attributes)
     statuses.create!(status_attributes)
     reload # to get the statuses into the right order (since sorted :desc)
     new_status = significant_status
-    Hook.emit(:commit_status, stack, commit: self, stack: stack, status: new_status) if previous_status != new_status
+    Hook.emit(:commit_status, stack, commit: self, stack: stack, commit_status: new_status,
+                                     status: new_status.state) if previous_status != new_status
+    if simple_state(previous_status) != simple_state(new_status) && new_status.state != 'pending'
+      Hook.emit(:deployable_status, stack, commit: self, stack: stack, deployable_status: new_status,
+                                           status: new_status.state)
+    end
```

Comment:
Not a huge deal breaker though. But I somehow find this block of code a bit too dense and slightly hard to read.